### PR TITLE
feat: Cookie authentication and corresponding middleware

### PR DESCRIPTION
**Pull request by RakeshUP**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Unit Test
- [ ] Integration Test

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -1,34 +1,48 @@
...
+import createError from 'http-errors';
+import express from 'express';
+import path from 'path';
+import mongoose from 'mongoose';
+import dotenv from 'dotenv';
+import requireAll from 'require-all';
+import pino from 'express-pino-logger';
+import cors from 'cors';
+import logger from './utils/logger';
+// import authorize from './thirdparty/routes/authorize';
```
Comment (rishabh1403) [inline server/app.js]: remove comments if not required :)

File: server/app.js [hunk 2]
```diff
@@ -1,34 +1,48 @@
...
+const publicFolder = path.join(__dirname, '../assets');
 
-app.use(pino);
+app.use(pino());
 app.use(express.json());
 app.use(express.urlencoded({ extended: false }));
-app.use(express.static(path.join(__dirname, 'public')));
+app.use(express.static(publicFolder));
+app.use('/api/v1', AuthenticationMiddleware);
+app.set('view engine', 'ejs');
```
Comment (rishabh1403) [inline server/app.js]: do you need a view engine?

File: server/constants/actionStatus.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+exports.NOT_ALLOWED = 'NOT ALLOWED';
```
Comment (rishabh1403) [inline server/constants/actionStatus.js]: like this :+1:

File: server/controller/UserController.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import response from '../lib/responseLib';
+import * as actionStatus from '../constants/actionStatus';
+import { hashPassword, comparePassword } from '../lib/passwordLib';
+import { createAuthToken } from '../lib/authTokenLib';
+
+const UsersCollection = mongoose.model('User');
+
+const createUserObject = async (email, password) => {
+  const userObject = {
+    email, password: await hashPassword(password),
```
Comment (rishabh1403) [inline server/controller/UserController.js]: use a separate line for assignment

File: server/controller/UserController.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  };
+  return userObject;
+};
+
+const createUser = async (req, res) => {
```
Comment (rishabh1403) [inline server/controller/UserController.js]: put line breaks, it a bit unreadbale

File: server/controller/UserController.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  const { email, password } = req.body;
+  const retrievedUser = await UsersCollection.findOne({ email });
```
Comment (rishabh1403) [inline server/controller/UserController.js]: what if `retrievedUser` is an empty object? Try to have strong checks

File: server/controller/UserController.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  const generatedResponse = response.generateResponse(false, actionStatus.SUCCESS, 'Signup Successful', { authToken });
+  res.cookie('authToken', authToken, { httpOnly: true, maxAge: 86400000 });
+  res.send(generatedResponse);
+};
+
+const signIn = async (req, res) => {
+  const { email, password } = req.body;
+  const retrievedUser = await UsersCollection.findOne({ email });
+  if (!retrievedUser) {
+    const generatedResponse = response.generateResponse(true, actionStatus.NOT_ALLOWED, 'Either Email or Password Is Wrong', null);
```
Comment (rishabh1403) [inline server/controller/UserController.js]: use objects as params when the number of params is more than 2-3

File: server/app.js (hunk 1 above)
Comment (RakeshUP) [reply server/app.js]: I have copied code from the hackathon branch, so it has all controllers and routes required for third party integration

File: server/app.js (hunk 2 above)
Comment (RakeshUP) [reply server/app.js]: I thought of serving the OAuth popup page from server and I have to have scripts in the page for closing the popup window

**Pull request merged**